source/speechDictHandler.py

```diff
@@ -21,6 +21,11 @@
 ENTRY_TYPE_WORD = 2 # String must have word boundaries on both sides to match
 ENTRY_TYPE_REGEXP = 1 # Regular expression
 
+#Types of regexp for parsing numbers:
+RE_SINGLE_DIGITS = re.compile(r"(\d)(?=\d+(\D|\b))")
+RE_DOUBLE_DIGITS = re.compile(r"(\d{1,2})(?=(\d{2})+(\D|\b))")
+RE_TRIPLE_DIGITS = re.compile(r"(\d{1,3})(?=(\d{3})+(\D|\b))")

```

Comment:
You probably want to include the `re.UNICODE` flag for these so that they match digits for other locales.